phys2bids/interfaces/txt.py

```diff
@@ -82,3 +83,164 @@ def populate_phys_input(filename, chtrig):
+        raise AttributeError(f'Interval unit "{interval[-1]}" is not in a '
+                             'valid AcqKnowledge format time unit, this probably'
+                             'means your file is not in min, sec, msec, µsec, Mhz, KHz or Hz')
+    interval[-1] = interval[-1].split('/')[0]
+    if 'Hz' in interval[-1].split('/')[0]:
+        print('frequency is given in the header, calculating sample Interval'
+              ' and standarizing to Hz if needed')
+        freq = float(interval[0])
+        freq_unit = interval[-1]
+        if freq_unit == 'MHz':
+            freq = freq * (1000000)
+        elif freq_unit == 'kHz':
+            freq = freq * 1000
+        interval[0] = 1 / freq
+        freq = [freq] * (len(timeseries) + 1)
+    else:
+        if interval[-1].split('/')[0] != 'sec':
+            print('Interval is not in seconds. Converting its value.')
+            if interval[-1].split('/')[0] == 'min':
+                interval[0] = float(interval[0]) * 60
+                interval[-1] = 's'
+            elif interval[-1].split('/')[0] == 'msec':
+                interval[0] = float(interval[0]) / 1000
+                interval[-1] = 's'
+            elif interval[-1].split('/')[0] == 'µsec':
+                interval[0] = float(interval[0]) / 1000000
+                interval[-1] = 's'
+        else:
+            interval[0] = float(interval[0])
+            interval[-1] = 's'
+        freq = [1 / interval[0]] * (len(timeseries) + 1)
+    # get units and names
+    orig_units = []
+    orig_names = []
+    # the for loop starts at index1 at 3 because that's the first line of the header
+    # with channel name info and ends in 2 + twice the number of channels because
+    # that should be the last channel name
+    for index1 in range(3, 3 + len(header[-1]) * 2, 2):
+        orig_names.append(header[index1][0])
+        # since units are in the line imediately after we get the units at the same time
+        orig_units.append(header[index1 + 1][0])
+    # reorder channels names
+    names = ['time', 'trigger']
+    orig_names.pop(chtrig - 1)
+    names = names + orig_names
+    # reoder channels units
+    units = ['s', 'Volts']
+    orig_units.pop(chtrig - 1)
+    units = units + orig_units
+    # get channels
+    timeseries = [np.array(darray) for darray in timeseries]
+    duration = (timeseries[0].shape[0] + 1) * interval[0]
+    t_ch = np.ogrid[0:duration:interval[0]][:-1]  # create time channel
+    ordered_timeseries = [t_ch, timeseries[chtrig - 1]]
+    timeseries.pop(chtrig - 1)
+    ordered_timeseries = ordered_timeseries + timeseries
+    return BlueprintInput(ordered_timeseries, freq, names, units)
+
+
+def read_header_and_channels(filename, chtrig):
+    header = []
+    channel_list = []
+    with open(filename, 'r') as f:
+        for line in f:
+            line = line.rstrip('\n').split('\t')
+            if line[-1] == '':
```

Comment:
```suggestion
            if line[-1] == ' ':
```

Reply:
Shouldn't it be like this if there's a final space?

Reply:
No, what happens is that due to the extra ``/t`` there is an extra field in line that is blank so I just eliminate it